### DISCUSSION TITLE
sign_bit function for float, double

### DISCUSSION THREAD

**Discussion by vbmithr** [duckdb/duckdb#5846] [Ideas]
Currently as far as I can see there is no way to know, for floats, if they have their sign bit set or not. It seems duckdb treats 0.0 and -0.0 the same way (sign(0.0) = sign(-0.0) = 0).

I have a use case where I store order book updates for trading data, where both 0.0 and -0.0 have a meaning, and I'd like to be able to use the information but right now I don't see a way.

**Comment by Mytherin** [comment]
We could indeed add the [signbit](https://en.cppreference.com/w/cpp/numeric/math/signbit) function to DuckDB. Happy to review a PR!

**Reply by carlopi** [reply]
Update: support for signbit just landed, look here for a basic test on how to use this: https://github.com/duckdb/duckdb/blob/master/test/sql/function/numeric/test_sign_bit.test
